### PR TITLE
Bound team calendar feed game reads

### LINKED ISSUE #3782: Bound game reads for team calendar feed endpoints
## Summary
Teams with multi-season schedules pay an unbounded Firestore read cost every time Apple/Google/public calendar clients poll a feed, which can slow feed responses and burn quota as teams, seasons, and subscribers grow.

## Workflow
Parent/fan calendar subscription feed generation for private team feeds and public team game feeds

## Root Cause
Calendar feed HTTP functions query the entire team games subcollection and filter/format in memory instead of applying a bounded date window suitable for calendar subscriptions.

## Recommendation
Add shared calendar-feed date window constants/helper and query both publicTeamGamesIcs and teamCalendarFeed with where('date','>=', start) and where('date','<=', end) plus orderBy('date'), preserving existing ICS formatting and token/public access checks.

## Evidence
- functions/index.js:publicTeamGamesIcs reads firestore.collection(`teams/${teamId}/games`).get() and then filters public games in memory, so every public feed request scans all historical games for that team.
- functions/index.js:teamCalendarFeed reads firestore.collection(`teams/${teamId}/games`).orderBy('date').get() before building ICS, so every private subscribed calendar poll scans the full team game history.
- apps/app/src/lib/parentCalendarService.ts:getPrivateTeamCalendarFeedUrl builds URLs for teamCalendarFeed, making this endpoint the app-backed private calendar subscription path.
- apps/app/src/pages/parent-tools/CalendarTool.tsx:openFeed sends parents to Apple/Google calendar subscription links, where recurring calendar-client polling multiplies the unbounded read pattern by subscribers.
- tests/unit/team-calendar-feed.test.js currently asserts the private feed uses orderBy('date').get(), proving the unbounded behavior is intentional enough to need a test change.

## Acceptance Criteria
- Private teamCalendarFeed no longer contains an unbounded orderBy('date').get() or collection get for games; it uses a bounded date-range query.
- publicTeamGamesIcs no longer reads all team games before filtering public games; it uses the same bounded date-range query before public visibility filtering.
- Calendar feeds still include near-past and future scheduled games inside the configured window and exclude out-of-window historical games.
- Existing private token validation, public visibility filtering, RSVP-summary formatting, and no-RSVP-subcollection-scan guarantees remain intact.

### Unit
- Update tests/unit/team-calendar-feed.test.js to assert bounded Firestore date constraints and to reject unbounded games collection scans.
- Add/update a public calendar function source contract test to assert publicTeamGamesIcs uses the bounded query helper before buildPublicGamesIcs.
- Add helper-level tests for the calendar feed window boundaries if the date-window helper is extracted.

### Integration
- Run the relevant Vitest function/unit suites for team calendar feed and public calendar core.
- Use mocked Firestore query objects or source-contract assertions to verify old, in-window, and future game docs are handled without reading RSVP/player subcollections.

### Smoke / Manual
- Manual bounded validation: seed or mock a team with old historical games plus current/future games, request teamCalendarFeed and publicTeamGamesIcs, and verify only in-window events appear in the ICS while endpoints return 200 for valid tokens/public teams.

## Non-Goals
- Do not redesign calendar token issuance or subscription URLs.
- Do not add paid observability, CDN caching, or a new persistence layer.
- Do not change RSVP summary calculation or hydrate RSVP/player subcollections in feed generation.

## Likely Files
- functions/index.js
- tests/unit/team-calendar-feed.test.js
- tests/unit/public-calendar-core.test.js

## Deduplication Notes
Checked GitHub issues for calendar fetch/cache/rate-limit and teamCalendarFeed/orderBy. Related closed issues #1820 and #1889 covered external ICS fetch caching, and #3215 removed private feed N+1 RSVP/player reads. This appears distinct because the current code still performs one full team games collection scan per feed request, and the existing unit test asserts that unbounded query.

---
Detected by `scale and rate-limit scout` using `codex-core` / Codex 5.5. Scoped for one focused PR.

### PR DESCRIPTION
**Pull request by pauljsnider**
## Summary

- Add a shared calendar-feed games query with a 90-day lookback and 365-day lookahead.
- Use the bounded query for both `publicTeamGamesIcs` and `teamCalendarFeed`.
- Add helper-level boundary/query tests and endpoint source contracts that reject the prior unbounded reads.

## Root cause

Both calendar subscription endpoints read the full team games subcollection on every poll. Public feeds filtered visibility only after that full read, while private feeds ordered the full collection without applying any date bounds.

## Impact

Recurring Apple, Google, and public calendar polls now read only recent and upcoming schedule documents. Existing public visibility filtering, private token/access validation, RSVP-summary behavior, event mapping, and ICS formatting remain unchanged.

## Validation

- `npx vitest run tests/unit/calendar-feed-window.test.js tests/unit/team-calendar-feed.test.js tests/unit/public-calendar-core.test.js --environment node --reporter=verbose` (15 tests passed)
- `npm test --prefix functions` (60 tests passed)
- `node --check functions/calendar-feed-window-core.cjs`
- `node --check functions/index.js`

Closes #3782

### CODE REVIEW

File: functions/calendar-feed-window-core.cjs [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  const anchor = now instanceof Date ? new Date(now.getTime()) : new Date(now);
+  if (Number.isNaN(anchor.getTime())) {
+    throw new TypeError('Calendar feed window requires a valid date');
+  }
+
+  return {
+    start: new Date(anchor.getTime() - CALENDAR_FEED_LOOKBACK_DAYS * MILLIS_PER_DAY),
+    end: new Date(anchor.getTime() + CALENDAR_FEED_LOOKAHEAD_DAYS * MILLIS_PER_DAY)
+  };
+}
```
[the remark this answers is not in the document]
Comment (pauljsnider) [reply functions/calendar-feed-window-core.cjs:15]: This suggestion is not applicable to this diff. calendar-feed-window-core.cjs has no teamId input or numeric-ID validation: line 15 closes getCalendarFeedDateWindow, whose only input is a date, and buildCalendarFeedGamesQuery accepts a Firestore collection reference plus an optional date. Team IDs remain path strings at the endpoint call sites and are not processed by this helper. No code change is needed for this thread.

**Comment by pauljsnider** [comment]
Reviewed current head `e43b94b653e0`.

paulbot-current-head-review: no blocking findings

│
◇  Doctor warnings ───────────────────────────────────────────────────────╮
│                                                                         │
│  - Left migrated task flow sidecar in place because archive already     │
│    exists:                                                              │
│    /home/paul-bot1/.openclaw-codex-core/flows/registry.sqlite.migrated  │
│                                                                         │
├─────────────────────────────────────────────────────────────────────────╯
paulbot-current-head-review: no blocking findings

Reviewed the current head against `origin/master` for the calendar feed query bounding change. No P0/P1/P2 regressions found: auth checks remain ahead of private feed reads, public visibility filtering remains intact, the bounded Firestore query matches existing timestamp-based schedule query patterns, and focused helper/source-contract tests cover the prior unbounded reads. Syntax and diff checks passed; Vitest could not start in this review lane because `vitest/config` was not resolvable.
⚠️ 🛠️ `npx vitest run tests/unit/calendar-feed-window.test.js test…-calendar-core.test.js --environment node --reporter=verbose (in ~/.local/state/paul-bot1/pr-lanes/head-review-workspaces/pauljsnider__allplays-pr-3794)` failed
